onnx/backend/test/cpp/driver_test.cc

```diff
@@ -0,0 +1,138 @@
+              onnxTensorDescriptorV1 weightDescriptors =
+                  ProtoToOnnxTensorDescriptor(model_.graph().initializer(0));
+
+              EXPECT_EQ(
+                  lib.onnxInitGraph(
+                      backend,
+                      NULL,
+                      sizeof(model_),
+                      &model_,
+                      weightCount,
+                      &weightDescriptors,
+                      &graph),
+                  ONNXIFI_STATUS_SUCCESS);
+
+              for (auto proto_test_data : protos_) {
+                std::vector<onnxTensorDescriptorV1> input_descriptor,
+                    output_descriptor, result_descriptor;
+                for (auto input : proto_test_data.inputs_) {
+                  input_descriptor.push_back(
+                      ProtoToOnnxTensorDescriptor(input));
+                }
+                for (auto output : proto_test_data.outputs_) {
+                  output_descriptor.push_back(
+                      ProtoToOnnxTensorDescriptor(output));
+                  onnxTensorDescriptorV1 result;
+                  result.tag = ONNXIFI_TAG_TENSOR_DESCRIPTOR_V1;
+                  result.name = "result";
+                  result.dataType = output.data_type();
+                  result.memoryType = ONNXIFI_MEMORY_TYPE_CPU;
+                  result.dimensions = output.dims().size();
+                  result.shape = (uint64_t*) output.dims().data();
+                  result.buffer =
+                      (onnxPointer) new char[sizeof(output.raw_data().size())];
+                  result_descriptor.push_back(result);
+                }
+                EXPECT_EQ(
+                    lib.onnxSetGraphIO(
+                        graph,
+                        input_descriptor.size(),
+                        input_descriptor.data(),
+                        result_descriptor.size(),
+                        result_descriptor.data()),
+                    ONNXIFI_STATUS_SUCCESS);
+
+                onnxMemoryFenceV1 inputFence, outputFence;
+                inputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+                outputFence.tag = ONNXIFI_TAG_MEMORY_FENCE_V1;
+                inputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+                outputFence.type = ONNXIFI_SYNCHRONIZATION_EVENT;
+
+                // EXPECT_EQ(lib.onnxInitEvent(backend, inputFence),
+                // ONNXIFI_STATUS_SUCCESS);
+                EXPECT_EQ(
+                    lib.onnxRunGraph(graph, &inputFence, &outputFence),
+                    ONNXIFI_STATUS_SUCCESS);
+                EXPECT_EQ(
+                    lib.onnxWaitEvent(outputFence.event),
+                    ONNXIFI_STATUS_SUCCESS);
+                for (int i = 0; i < output_descriptor.size(); i++) {
+                  auto output_size = getDescriptorSize(&output_descriptor[i]);
+                  for (int j = 0; j < output_size; j++) {
+                    // size might be a problem!
+                    EXPECT_EQ(
```

Comment:
This is not good. We cannot compare them byte by byte because there might be numerical noise.

Reply:
yes, I agree with you, thats why I set a reminder here... wonder if you have any suggestions?